server/gql/models/tests/authors/authors.test.js

```diff
@@ -0,0 +1,44 @@
+import get from 'lodash/get';
+import { graphqlSync, GraphQLSchema } from 'graphql';
+import { createFieldsWithType, expectSameTypeNameOrKind } from '@utils/testUtils';
+import { QueryRoot } from '@server/gql/queries';
+import { MutationRoot } from '@server/gql/mutations';
+import { authorsFields } from '@gql/models/authors';
+import { timestamps } from '@gql/models/timestamps';
+
+const schema = new GraphQLSchema({ query: QueryRoot, mutation: MutationRoot });
+
+let fields = [];
+
+fields = createFieldsWithType({ ...authorsFields, ...timestamps });
+
+const query = `
+  {
+    __type(name: "Author") {
+        name
+        kind
+        fields {
+          name
+          type {
+            name
+            kind
+          }
+        }
+      }    
+  }
+`;
+describe('Author introspection tests', () => {
+  it('should have the correct fields and types', () => {
+    const result = graphqlSync({ schema, source: query });
+    const authorFieldTypes = get(result, 'data.__type.fields');
+    const hasCorrectFieldTypes = expectSameTypeNameOrKind(authorFieldTypes, fields);
+    expect(hasCorrectFieldTypes).toBeTruthy();
```

Comment:
lets check the actual values as well

Reply:
roger

Reply:
similar to what you're doing [here](https://github.com/sagar-wednesday/node-express-graphql-template/pull/17/files#diff-19725ff9c6126680d6e9ba1ffb7428a9859e75406b534da2f2d108ea0a7e7792R56) this needs to be checked as well